File: Example4OC/Example4OC/Classes/Map/BVMapController.m
//
//  BVMapController.m
//  LibosExample
//
//  Created by 黑眸智能 on 2021/11/2.
//

#import "BVMapController.h"
#import "BVRegular.h"

@interface BVMapController ()



@end

@implementation BVMapController

- (void)viewDidLoad {
    [super viewDidLoad];
    [self initView];
}

- (void)dealloc {
    NSLog(@"销毁");
}

-(NSString *)getNavTitle {
    return @"地图显示";
}

-(BVSweeperMapViewMode)getMapMode {
    return BVSweeperMapViewModeNormalMode;
}

-(void)initView {
    self.navigationItem.title = [self getNavTitle];
    self.view.backgroundColor = UIColor.systemBackgroundColor;
    
    UIView *emptyView = [[UIView alloc]init];
    emptyView.backgroundColor = UIColor.redColor;
    
    BVSweeperMapViewMode mode = [self getMapMode];
    
    CGFloat height = [AppDelegate getInstance].window.windowScene.statusBarManager.statusBarFrame.size.height;
    
    BVSweeperMapView *mapView = [[BVSweeperMapView alloc]initWithMode:mode emptyView:emptyView];
    mapView.backgroundColor = UIColor.grayColor;
    [self.view addSubview:mapView];
    [mapView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.right.equalTo(self.view);
        make.top.equalTo(self.view).offset(height);
        make.height.mas_equalTo(500);
    }];
    
    UITextView *logView = [[UITextView alloc]init];
    [self.view addSubview:logView];
    [logView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.left.right.equalTo(self.view);
        make.top.equalTo(mapView.mas_bottom);
        make.height.mas_equalTo(150);
    }];
    
    NSData *data = [BVRegular loadData];
    BVSweeperMapData *model = [BVSweeperMapData modelWithData:data];
    [mapView reciveMapData:model];
    
    self.mapView = mapView;
    self.logView = logView;
}

-(void)log:(NSString *)msg {
    NSString *text = [NSString stringWithFormat:@"%@\n%@",self.logView.text,msg];
    self.logView.text = text;
}

@end

File: Example4Swift/Example4Swift/Classes/Map/BVMapController.swift
//
//  BVMapController.swift
//  Libos
//
//  Created by 田宇 on 2021/10/28.
//

import UIKit
import BVSweeperKit

class BVMapController: UIViewController {
    
    var mapView: BVSweeperMapView!
    var logView: UITextView!

    override func viewDidLoad() {
        super.viewDidLoad()
        initView()
    }
    
    deinit {
        print("销毁")
    }
    
    
}

@objc extension BVMapController {
    
    func getNavTitle() -> String {
        return "地图显示"
    }
    
    func getMapMode() -> BVSweeperMapViewMode {
        .normalMode
    }
    
    func initView() {
        self.navigationItem.title = getNavTitle()
        
        self.view.backgroundColor = .secondarySystemBackground
        
        let emptyView = UIView.init()
        emptyView.backgroundColor = .red
        
        let mode = getMapMode()
        
        let height = AppDelegate.default.window?.windowScene?.statusBarManager?.statusBarFrame.height ?? 20
        
        mapView = BVSweeperMapView.init(mode: mode, emptyView: emptyView)
        mapView.backgroundColor = .systemGray
        self.view.addSubview(mapView)
        mapView.snp.makeConstraints { make in
            make.left.right.equalToSuperview()
            make.top.equalToSuperview().offset(height)
            make.height.equalTo(500)
        }
        
        logView = UITextView.init()
        logView.isEditable = false
        self.view.addSubview(logView)
        logView.snp.makeConstraints { make in
            make.left.right.equalToSuperview()
            make.top.equalTo(mapView.snp.bottom)
            make.height.equalTo(150)
        }
        
        if let data = BVRegular.loadData() {
            if let mapData = BVSweeperMapData.modelWith(data: data) {
                self.mapView.reciveMapData(mapData)
            }
        }
    }
    
    func log(_ msg: String) {
        var text = self.logView.text
        text?.append("\n")
        text?.append(msg)
        self.logView.text = text
    }
}
